server/apps/immich/src/api-v1/config/config.controller.ts

```diff
@@ -0,0 +1,34 @@
+import { BadRequestException, Body, Controller, Get, Put, UseGuards, ValidationPipe } from '@nestjs/common';
+import { ConfigService } from './config.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { JwtAuthGuard } from '../../modules/immich-jwt/guards/jwt-auth.guard';
+import { AdminRolesGuard } from '../../middlewares/admin-role-guard.middleware';
+import { AdminConfigResponseDto } from './response-dto/admin-config-response.dto';
+import { SetAdminConfigDto } from './dto/set-admin-config';
+import { AdminConfigKey } from '@app/database/entities/admin-config.entity';
+
+@UseGuards(JwtAuthGuard)
+@UseGuards(AdminRolesGuard)
+@ApiTags('Admin Config')
+@ApiBearerAuth()
+@Controller('config')
+export class ConfigController {
+  constructor(private readonly adminConfigService: ConfigService) {}
+
+  @Get('/admin')
```

Comment:
I'd maybe just have this be `@Get()`, unless you are thinking of having different types of configurations added here in the future? If that's the case I don't think "admin" makes sense as a configuration type. Maybe "system" instead?

Reply:
Yea the idea behind this was that I wanted to have a priviledged config while still using the /config namespace, happy to go with system instead, would require a bit of refactoring to move everything to system config